src/ast/comparable.rs

```diff
@@ -424,7 +424,11 @@ impl<'a> From<&'a Expr> for ComparableExpr<'a> {
                 orelse: orelse.into(),
             },
             ExprKind::Dict { keys, values } => Self::Dict {
-                keys: keys.iter().map(std::convert::Into::into).collect(),
+                keys: keys
+                    .iter()
+                    .flatten()
+                    .map(std::convert::Into::into)
+                    .collect(),
```

Comment:
I think this may need to preserve the `None`, to be truly comparable, right?

Reply:
(So we'd need to update the definition of `ComparableExpr`.)

Reply:
(I'll go ahead and do this, then merge.)

Reply:
thanks for the update!